### DISCUSSION TITLE
Is `getCurrentInstance()` safe to use in plugin development?

### DISCUSSION THREAD

**Discussion by Rogue3799** [vueuse/vueuse#2564] [Q&A]
Curious as to your thoughts on using `getCurrentInstance()`. I've been trying to figure out whether or not I should use it in the plugin I'm developing. Is it safe to use even though it's an internal API that's not in the Vue documentation?

See my discussion on Vue repo, if necessary:
https://github.com/vuejs/docs/discussions/2134#discussion-4680445

https://github.com/vueuse/vueuse/blob/4dea4be6f46a85a880ccd08954c84febf332187a/packages/core/templateRef/index.ts#L16

**Comment by LockingReal** [comment]
> 想知道您对使用`getCurrentInstance()`. 我一直在想我是否应该在我正在开发的插件中使用它。即使它是不在 Vue 文档中的内部 API，使用它是否安全？
> 
> 如有必要，请参阅我对 Vue repo 的讨论： [vuejs/docs#2134（评论）](https://github.com/vuejs/docs/discussions/2134#discussion-4680445)
> 
> https://github.com/vueuse/vueuse/blob/4dea4be6f46a85a880ccd08954c84febf332187a/packages/core/templateRef/index.ts#L16

https://stackoverflow.com/questions/72209080/vue-3-is-getcurrentinstance-deprecated

I think the answer in this post iskvery good. The api should be used with caution. But in hook, I think we should not use a variable with context that is passed in by non-user parameters, except that it is provided by the browser
